processing/src/main/java/io/druid/segment/serde/ComplexColumnPartSerde.java

```diff
@@ -58,9 +68,25 @@ public String getTypeName()
     return typeName;
   }
 
-  public static SerializerBuilder serializerBuilder()
+  @Override
+  public Serializer getSerializer()
   {
-    return new SerializerBuilder();
+    return serializer;
+  }
+
+  @Override
+  public Deserializer getDeserializer()
+  {
+    return new Deserializer()
+    {
+      @Override
+      public void read(ByteBuffer buffer, ColumnBuilder builder, ColumnConfig columnConfig)
+      {
+        if (serde != null) {
```

Comment:
Keeping the same behavior as before  https://github.com/druid-io/druid/pull/3743/files#diff-c28c34e2beb6a4aa2a04d2c26ba07bfdL162 .